src/main/java/org/jahia/modules/sitemap/services/impl/SitemapServiceImpl.java

```diff
@@ -179,17 +196,36 @@ public String getSitemap(String siteKey, String key) {
     }
 
     @Override
-    public void addSitemap(String siteKey, String key, String sitemap) throws RepositoryException {
-        JCRTemplate.getInstance().doExecuteWithSystemSession(session -> {
+    public void addSitemap(String siteKey, String key, Path sitemap) throws RepositoryException {
+        JCRTemplate.getInstance().doExecuteWithSystemSession( session -> {
             JCRNodeWrapper settingsNode = JCRContentUtils.getOrAddPath(session, session.getNode("/sites/" + siteKey), "sitemapSettings", "jnt:sitemapSettings");
             JCRNodeWrapper cacheRoot = JCRContentUtils.getOrAddPath(session, settingsNode, "sitemapCache", "jnt:sitemapRootCacheEntries");
             JCRNodeWrapper cacheNode = JCRContentUtils.getOrAddPath(session, cacheRoot, key, "jnt:sitemapEntry");
-            cacheNode.setProperty("sitemap", sitemap);
+            try(FileInputStream stream = new FileInputStream(sitemap.toFile())) {
+                cacheNode.setProperty("sitemapFile", session.getValueFactory().createBinary(stream));
+            } catch (Throwable e) {
+                throw new RuntimeException(e);
+            }
             session.save();
             return null;
         });
     }
 
+    private void removeSitemaps() {
```

Comment:
This function has been removed in the previous commit https://github.com/Jahia/sitemap/pull/192/files#diff-ba5fe80985eb4b164cfe8bfede2fb565d9d23c3f50c91de70101365ba3b6be1dL198.
You maybe missed this when doing the rebase

Reply:
I missed that one while rebasing .. true